pepper-apis/src/main/java/org/broadinstitute/ddp/service/WorkflowService.java

```diff
@@ -59,7 +69,12 @@ public WorkflowService(PexInterpreter interpreter) {
             }
         }
 
-        return Optional.ofNullable(next);
+        Optional<WorkflowState> nextState = Optional.ofNullable(next);
+        nextState.ifPresent(nextWfState -> {
+            createActivityInstanceIfMissing(handle, fromState, nextWfState, operatorGuid, userGuid);
+        });
+
+        return nextState;
     }
 
     public WorkflowResponse buildStateResponse(Handle handle, String userGuid, WorkflowState state) {
```

Comment:
Now that we're always returning an `instanceGuid`, do you think it makes sense to include the instance guid directly in the `ActivityState` object? So we don't have to do another query to find it in `buildStateResponse()`.

Reply:
Yes, this optimization seemed natural after the changes were made to `WorkflowService` and avoiding one more DB trip is always nice. However, I tinkered with it a bit and got a feeling that it doesn't worth the trouble - too many changes for a small gain. If you insist, let's do that, but I'd leave it as-is. @zyme, what do you think?

Reply:
We can refactor separately.